File: tests/py/testPolymorph.py
import swigex as mf

#pa = mf.AParent() # AParent is a pure abstract class
#mf.showHello(pa)

co = mf.ChildOne()
mf.showHello(co)

# This class is also pure abstract (should override clone)
# But it works !!!
class ChildTwo(mf.AParent):
    def __init__(self):
        super(ChildTwo,self).__init__()
    
    def getHello(self):
        return "ChildTwo in Python - Hello"
    
    #def clone(self):
    #    return ChildTwo()

ct = ChildTwo()
mf.showHello(ct)

File: tests/r/testPolymorph.r
library(swigex)

#pa = AParent() # AParent is a pure abstract class
#invisible(showHello(pa))

co = ChildOne()
invisible(showHello(co))

# TODO : Cannot inherits from AParent in R
#ct = AParent()
#class(ct) <- c("ChildTwo", "AParent")
#getHello.ChildTwo <- function(obj)
#{
#  return("ChildTwo - Hello")
#}
#
#showHello(ct)
